### ISSUE TITLE
IntelliSence incorrectly shows problem when calling consteval function "cannot call non-constexpr function"

### ISSUE THREAD

**Issue by hsandt** [microsoft/vscode-cpptools#5627]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Linux Mint 19.2
- VS Code Version: 1.45.1
- C/C++ Extension Version: 0.28.3
- Other extensions you installed (and if the issue persists after disabling them): several, but disabling them didn't change anything (they were not related to C++)
- Does this issue involve using SSH remote to run the extension on a remote machine?: no
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

**Steps to reproduce**

1. Create a folder with file main.cpp
2. Open the folder in VS code (to enable workspace configuration)
3. In main.cpp paste this:

```cpp
consteval int sqr(int n) {
  return n*n;
}

int main(int argc, char const *argv[])
{
    constexpr int r = sqr(100);  // OK

    return 0;
}
```

At this point your project should still be set for C++11, so consteval will be underlined as an error.

4. Use command C/C++: Edit Configurations (UI) and set C++ standard to C++20

5. See Intellisense error: "call to consteval function "sqr" did not produce a valid constant expression -- cannot call non-constexpr function "sqr" (declared at line 1)"

**Expected behavior**
No Intellisense error

<details>
  <summary><strong>Logs</strong></summary>

```
-------- Diagnostics - 6/9/2020, 7:59:27 PM
Version: 0.28.3
Current Configuration:
{
    "name": "Linux",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [],
    "compilerPath": "/usr/bin/clang",
    "cStandard": "c11",
    "cppStandard": "gnu++20",
    "intelliSenseMode": "clang-x64"
}
Translation Unit Mappings:
[ .../main.cpp ]:
    .../main.cpp
Translation Unit Configurations:
[ .../main.cpp ]:
    Process ID: 15759
    Memory Usage: 23 MB
    Compiler Path: /usr/bin/clang
    Includes:
        /usr/include/c++/7
        /usr/include/x86_64-linux-gnu/c++/7
        /usr/include/c++/7/backward
        /usr/local/include
        /usr/lib/llvm-10/lib/clang/10.0.0/include
        /usr/include/x86_64-linux-gnu
        /usr/include
    Standard Version: c++20
    IntelliSense Mode: clang-x64
    Other Flags:
        --clang
        --clang_version=100000
Total Memory Usage: 23 MB
```

C++ output
```
idle loop: reparsing the active document
Checking for syntax errors: file:///.../main.cpp
Queueing IntelliSense update for files in translation unit of: .../main.cpp
Error squiggle count: 0
Update IntelliSense time (sec): 0.255
  tag parsing file: .../main.cpp
sending 1 changes to server
sending 1 changes to server
idle loop: reparsing the active document
Checking for syntax errors: file:///.../main.cpp
Queueing IntelliSense update for files in translation unit of: .../main.cpp
Error squiggle count: 1
Update IntelliSense time (sec): 0.275
Checking for syntax errors: file:///.../main.cpp
Queueing IntelliSense update for files in translation unit of: .../main.cpp
Error squiggle count: 1
Update IntelliSense time (sec): 0.255
```
</details>

**Screenshots**

![image](https://user-images.githubusercontent.com/1067772/84185874-911b1700-aa87-11ea-9555-bd17d09b5399.png)

**Additional context**

**Comment by sean-mcmanus** [comment]
Can you provide a code sample repro?

Also, what info is shown when you run C/C++: Log Diagnostics? i.e. what is the intelliSenseMode, cppStandard, etc.?

**Comment by hsandt** [comment]
Done. My browser bugged and submitted the issue before I could fill it. That's why I also had to rename that weird title.

**Comment by sean-mcmanus** [comment]
It will be fixed in our next 0.29.0-insiders release (not sure yet when it'll be available).

**Comment by sean-mcmanus** [comment]
Fixed with 0.29.0.
